core/common/src/main/java/alluxio/network/netty/NettyRPC.java

```diff
@@ -77,6 +72,32 @@ public void operationComplete(ChannelFuture future) throws Exception {
     return message;
   }
 
+  /**
+   * Sends a request and waits until the request is flushed to network. The caller of this method
+   * should expect no response from the server and hence the service handler on server side should
+   * not return any response or there will be issues. This method is typically used for RPCs
+   * providing best efforts (e.g., async cache).
+   *
+   * @param context the netty RPC context
+   * @param request the RPC request
+   */
+  public static void fireAndForget(final NettyRPCContext context, ProtoMessage request)
+      throws IOException {
+    Channel channel = Preconditions.checkNotNull(context.getChannel());
+    channel.writeAndFlush(new RPCProtoMessage(request)).addListener((ChannelFuture future) -> {
+      if (future.cause() != null) {
+        future.channel().close();
+      }
+      request.notify();
+    });
+    try {
+      request.wait();
```

Comment:
Is the ordering here guaranteed, could the call back fire before we enter `request.wait`?